agent/xds/routes.go

```diff
@@ -188,20 +231,13 @@ func makeUpstreamRouteForDiscoveryChain(
 		panic("unknown first node in discovery chain of type: " + startNode.Type)
 	}
 
-	return &envoy.RouteConfiguration{
-		Name: routeName,
-		VirtualHosts: []envoyroute.VirtualHost{
-			envoyroute.VirtualHost{
-				Name:    routeName,
-				Domains: []string{"*"},
-				Routes:  routes,
-			},
-		},
-		// ValidateClusters defaults to true when defined statically and false
-		// when done via RDS. Re-set the sane value of true to prevent
-		// null-routing traffic.
-		ValidateClusters: makeBoolValue(true),
-	}, nil
+	host := &envoyroute.VirtualHost{
```

Comment:
We could return `envoyroute.VirtualHost` instead, since the caller is dereferencing the pointer to construct an `[]envoyroute.VirtualHost`.

Reply:
Done